test/purview_test.tsx

```diff
@@ -1905,6 +1908,72 @@ test("reconnect early", async () => {
   })
 })
 
+test("cleanup happens when 'close' and 'connect' race", async () => {
+  let mountCount = 0
+  let unmountCount = 0
+
+  class Foo extends TestComponent<{}, { text: string }> {
+    state = { text: "hi" }
+
+    constructor(props: {}) {
+      super(props)
+    }
+
+    componentDidMount(): void {
+      mountCount++
+    }
+
+    componentWillUnmount(): void {
+      unmountCount++
+    }
+
+    render(): JSX.Element {
+      return <p>{this.state.text}</p>
+    }
+  }
+
+  await renderAndConnect(<Foo />, async conn => {
+    expect(mountCount).toBe(1)
+    expect(unmountCount).toBe(0)
+    conn.ws.close()
+
+    // Wait for state to be saved and unmount to occur. And wait for the state
+    // to be saved into the db to help simulate the page refresh scenario.
```

Comment:
Comment seems a bit redundant because the first sentence already mentions waiting for the state to be saved